### ISSUE TITLE
HttpServletRequest#getServerName always returns 0.0.0.0 when using new performant HTTP connector

### ISSUE THREAD

**Issue by ryantgraham** [GoogleCloudPlatform/appengine-java-standard#325]
We've found what we believe to be an issue where calls to `HttpServletRequest#getServerName` when the [new performant HTTP connector](https://github.com/GoogleCloudPlatform/appengine-java-standard/blob/main/httpconnector.md) has been enabled is returning a different value when it is not enabled.

With the connector _not_ enabled the value would be along the lines of:
`{version}.appspot.com`
![image](https://github.com/user-attachments/assets/24762ad9-1cac-4e87-b124-936f5404fdde)

With the connector enabled it always returns:
`0.0.0.0`
![image](https://github.com/user-attachments/assets/09cd13e3-ccd7-4abf-82d3-303aec9a2089)

This can be reproduced using the Google App Engine Standard Environment Samples for Java 17 Bundled Services [sample application](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/main/appengine-java17-bundled-services) and logging the output of the call to `HttpServletRequest#getServerName` in one of the servlets.

**Comment by ludoch** [comment]
Thanks for the report.

Maybe a regression in https://github.com/GoogleCloudPlatform/appengine-java-standard/blob/63b450dd18d6ed40550da57216691c9e9cf5433d/runtime/runtime_impl_jetty9/src/main/java/com/google/apphosting/runtime/jetty9/JettyRequestAPIData.java#L339 via https://github.com/GoogleCloudPlatform/appengine-java-standard/commit/a9e0ea529ff0983ad7171005df4da2374e149ce9 ?

**Comment by maigovannon** [comment]
@ludoch I think the issue might be the right one a54e790 since the https://github.com/GoogleCloudPlatform/appengine-java-standard/commit/a9e0ea529ff0983ad7171005df4da2374e149ce9 was a field change.

**Comment by lachlan-roberts** [comment]
Yes looks like its from https://github.com/GoogleCloudPlatform/appengine-java-standard/pull/311.

I will open a PR to fix the issue and increase the coverage from the test cases.

**Comment by ryantgraham** [comment]
Looks like this fix was included as part of https://github.com/GoogleCloudPlatform/appengine-java-standard/releases/tag/v2.0.32 Thanks so much for taking care it!
